src/autopas/containers/cellPairTraversals/SlicedBlkBasedTraversal.h

```diff
@@ -0,0 +1,953 @@
+   *      / |   210          / |   211          / |   212
+   *  020   |  / |       021   |  / |       022   |  / |
+   *   |   110   |        |   111   |        |   112   |
+   *   |  / |   200       |  / |   201       |  / |   202
+   *  010   |  /         011   |  /         012   |  /
+   *   |   100            |   101            |   102
+   *   |  /               |  /               |  /
+   *  000................001................002
+   */
+  std::vector<std::array<std::array<std::array<unsigned long, 2>, 3>, 27>> _allSubBlocks;
+
+  /**
+   * A map converting consecutive ints to the subBlock Order ints e.g. 000->0, 111-14, 201->11, ...
+   */
+  std::map<int, int> _subBlockOrderToSubBlockIndex;
+
+  /**
+   * A function mapping a vector of the corresponding subBlocks to the corresponding lock ids.
+   */
+  int lockToSubBlocks(int x, int y, int z) {
+    // 0 => my lock and 1 => neighbour lock
+    // x,y,z == 0 does not include 111 as this does not need a lock!
+    if (x == 0 && y == 0 && z == 0) return 0;
+    if (x == 1 && y == 0 && z == 0) return 100;
+    if (x == 0 && y == 1 && z == 0) return 10;
+    if (x == 0 && y == 0 && z == 1) return 1;
+
+    if (x == 1 && y == 1 && z == 0) return 110;
+    if (x == 1 && y == 0 && z == 1) return 101;
+    if (x == 0 && y == 1 && z == 1) return 11;
+    if (x == 1 && y == 1 && z == 1) return 111;
+  }
+
+  /**
+   * An array holding the locks in order of iteration over the subblocks of a block of cells. Is the same for all
+   * blocks/ threads.
+   */
+  std::array<std::array<unsigned long, 3>, 8> _cellBlockTraverseOrderByLocks;
+
+  /**
+   * A map converting the cellblock Coordinates (relative Coordinates iterated by Cellblock) to their index
+   */
+  std::map<std::array<unsigned long, 3>, unsigned long> _cellBlocksToIndex;
+  /**
+   * A map converting a cellblock index to their relative cellblock coordinates.
+   */
+  std::map<unsigned long, std::array<unsigned long, 3>> _indexToCellBlock;
+
+  std::string debugHelperFunctionIntArrayToString(std::array<std::vector<unsigned long>, 3> _cellBlockDimensions) {
+    std::string str;
+    for (int j = 0; j < 3; ++j) {
+      str += std::to_string(j) + ": ";
+      for (auto i : _cellBlockDimensions[j]) {
+        str += std::to_string(i) + "| ";
+      }
+    }
+    return str;
+  }
+  std::string debugHelperFunctionIntArrayToString(std::array<unsigned long, 3> intArray) {
+    std::string str;
+    for (auto i : intArray) {
+      str += std::to_string(i) + " ";
+    }
+    return str;
+  }
+  std::string debugHelperFunctionIntArrayToString2(std::array<unsigned long, 3> intArray) {
+    std::string str;
+    for (auto i : intArray) {
+      str += std::to_string(i) + " ";
+    }
+    return str;
+  }
+  std::string debugHelperFunctionIntArrayToString(std::array<double, 3> intArray) {
+    std::string str;
+    for (auto i : intArray) {
+      str += std::to_string(i) + " ";
+    }
+    return str;
+  }
```

Comment:
remove those. If they are really needed for some logging an equivalent function already exists in ArrayUtils